walletlibrary/src/main/java/com/microsoft/walletlibrary/util/http/URLFormEncoding.kt

```diff
@@ -0,0 +1,64 @@
+package com.microsoft.walletlibrary.util.http
+
+import java.io.Serializable
+import java.net.URLEncoder
+
+/**
+ *
+ */
+object URLFormEncoding {
+    const val mimeType = "application/x-www-form-urlencoded"
+
+    /***
+     * Encodes a map of string to  string, string?, or List<string> to URL Form Encoded payload
+     * @throws InvalidURLEncodeData if any other type is in the Map
+     */
+    fun encode(data: Map<String, Any?>): ByteArray {
+        val builder = StringBuilder()
+        var firstKeyPair = true
+        data.forEach { (key, value) ->
+            val encodedKey = URLEncoder.encode(key, "UTF-8")
+            when (value) {
+                is String -> {
+                    if (!firstKeyPair) {
```

Comment:
nit: move the code that actually performs URL encoding of string to a separate method.